### PR TITLE
community: Enhance Tencent Cloud VectorDB, langchain: make Tencent Cloud VectorDB self query retrieve compatible

### PR DESCRIPTION
**Pull request by jeffkit**
- make Tencent Cloud VectorDB support metadata filtering.
- implement delete function for Tencent Cloud VectorDB.
- support both Langchain Embedding model and Tencent Cloud VDB embedding model.
- Tencent Cloud VectorDB support filter search keyword,  compatible with langchain filtering syntax.
- add Tencent Cloud VectorDB TranslationVisitor, now work with self query retriever.
- more documentations.

### CODE REVIEW

**Comment by jeffkit** [comment]
@baskaryan errors were fixed, need your review again :P

File: docs/docs/integrations/retrievers/self_query/tencentvectordb.ipynb
Comment (baskaryan) [inline docs/docs/integrations/retrievers/self_query/tencentvectordb.ipynb:1]: seems to be an issue with the notebook

File: libs/community/langchain_community/vectorstores/tencentvectordb.py [hunk 1]
```diff
@@ -63,6 +79,63 @@ def __init__(
...
+            else:
+                raise ValueError(f"unsupported data_type {self.data_type}")
+        else:
+            if self.data_type not in enum.FieldType:
+                raise ValueError(f"unsupported data_type {self.data_type}")
+
+
+def is_valid_embedding(embedding: Embeddings) -> bool:
+    return embedding is not None and not isinstance(
+        embedding, (FakeEmbeddings, DeterministicFakeEmbedding)
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/tencentvectordb.py:112]: why do we need to check this?

File: docs/docs/integrations/retrievers/self_query/tencentvectordb.ipynb
Comment (jeffkit) [reply docs/docs/integrations/retrievers/self_query/tencentvectordb.ipynb:1]: fixed

File: libs/community/langchain_community/vectorstores/tencentvectordb.py (hunk 1 above)
Comment (jeffkit) [reply libs/community/langchain_community/vectorstores/tencentvectordb.py:112]: The purpose of FakeEmbedding is solely for demo purposes, and Tencent Cloud VectorDB allows the option not to specify LangChain Embeddings models (the alternative is to specify an embedding model name built into Tencent Cloud VectorDB). Therefore, we do not recommend explicitly using FakeEmbedding for demos, and thus classify FakeEmbedding as an illegal embedding.

If you believe it's necessary to keep the FakeEmbedding for demonstration purposes, then I can modify the code accordingly. What do you think?

Comment (baskaryan) [reply libs/community/langchain_community/vectorstores/tencentvectordb.py:112]: using fake embeddings can also be useful for testing purposes

Comment (jeffkit) [reply libs/community/langchain_community/vectorstores/tencentvectordb.py]: OK, fixed it.

Comment (jeffkit) [reply libs/community/langchain_community/vectorstores/tencentvectordb.py]: fixed all the CI issue, let's try again @baskaryan

Comment (jeffkit) [reply libs/community/langchain_community/vectorstores/tencentvectordb.py]: Is my fault，find_spec can throw a ModuleNotFoundError。 fixed it now. @baskaryan

File: libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py [hunk 2]
```diff
@@ -82,6 +82,7 @@ def check_compatibility(vector_store: VectorStore) -> bool:
         "SurrealDBStore",
         "TileDB",
         "TimescaleVector",
+        "TencentVectorDB",
```
Comment (baskaryan) [inline libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:85]: can we update docs/docs/modules/data_connection/indexing.ipynb to list TencentVectorDB

Comment (jeffkit) [reply libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:85]: sure, now it's in the Compatible Vectorstores list.

**Comment by jeffkit** [comment]
any other suggestions? @baskaryan

**Review by baskaryan — APPROVED** [review]

**Comment by jeffkit** [comment]
fixed the conflict on the indexing doc. need approval again @baskaryan

**Comment by jeffkit** [comment]
any other suggestions? @baskaryan

**Pull request merged**